caffe2/operators/conv_gradient_op.cc

```diff
@@ -4,8 +4,85 @@
 
 namespace caffe2 {
 
+std::vector<TensorShape> TensorInferenceForConvGradient(
+    const OperatorDef& def,
+    const std::vector<TensorShape>& in) {
+  CAFFE_ENFORCE(in.size() >= 2, "ConvGradient expects at least 2 inputs");
+
+  if (in[0].unknown_shape()) {
+    std::vector<TensorShape> out(1);
+    out[0].set_unknown_shape(true);
+    return out;
+  }
+  ArgumentHelper helper(def);
+  const auto no_bias = helper.GetSingleArgument<int>("no_bias", 0);
+  const auto n_outputs = def.output_size();
+  auto i = 0;
+  vector<TensorShape> out(n_outputs);
+
+  out[i++] = in[1];
```

Comment:
Maybe it's just me but i++ for indexing confuses me. Perhaps out[i] = in[1]; ++i; ?

``
  // FILTER_GRAD has the same shape as FILTER
  out[i] = in[1];
  ++i;
``